src/components/MapView/AlertForm/index.tsx

```diff
@@ -0,0 +1,354 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  Box,
+  Button,
+  createStyles,
+  FormControl,
+  ListSubheader,
+  MenuItem,
+  Select,
+  TextField,
+  Theme,
+  Typography,
+  withStyles,
+  WithStyles,
+} from '@material-ui/core';
+import { grey } from '@material-ui/core/colors';
+
+import { faBell, faCaretDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  getBoundaryLayerSingleton,
+  LayerDefinitions,
+} from '../../../config/utils';
+import {
+  BoundaryLayerProps,
+  WMSLayerProps,
+  LayerKey,
+} from '../../../config/types';
+import { LayerData } from '../../../context/layers/layer-data';
+import { layerDataSelector } from '../../../context/mapStateSlice/selectors';
+import { availableDatesSelector } from '../../../context/serverStateSlice';
+import LayerDropdown from '../Layers/LayerDropdown';
+
+const boundaryLayer = getBoundaryLayerSingleton();
+
+function AlertForm({ classes }: AlertFormProps) {
+  const dispatch = useDispatch();
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const availableDates = useSelector(availableDatesSelector);
+
+  const [isAlertFormFormOpen, setIsAlertFormFormOpen] = useState(false);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState<LayerKey>();
+  const [selectedDate, setSelectedDate] = useState<number | null>(null);
+  const [regionsList, setRegionsList] = useState<string[]>(['allRegions']);
+  const [emailValid, setEmailValid] = useState<boolean>(false);
+  const [email, setEmail] = useState<string>('');
+  const [selectedStat, setSelectedStat] = useState<string>('');
+  const [selectedThreshold, setSelectedThreshold] = useState<number>(0);
+
+  // Very basic regex for detecting emails
+  const emailRegex: RegExp = RegExp('.+@.+');
+
+  // On by default, may want to disable it in some cases
+  const isAlertFormEnabled = true;
```

Comment:
Constants should be at the top, outside of the functional component and (based off personal preference) cased as 'CAPITAL_SNAKE_CASE'